stream-chat-android-client/src/main/java/io/getstream/chat/android/client/setup/state/internal/ClientStateImpl.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.client.setup.state.internal
+
+import io.getstream.chat.android.client.models.ConnectionState
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.setup.state.ClientMutableState
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+internal class ClientStateImpl : ClientMutableState {
+
+    private val _initialized = MutableStateFlow(false)
+    private val _connectionState = MutableStateFlow(ConnectionState.OFFLINE)
+    private val _user = MutableStateFlow<User?>(null)
+
+    override val user: StateFlow<User?> = _user
+
+    override val initialized: StateFlow<Boolean> = _initialized
+
+    override val connectionState: StateFlow<ConnectionState> = _connectionState
+
+    override fun isOnline(): Boolean = _connectionState.value == ConnectionState.CONNECTED
+
+    override fun isOffline(): Boolean = _connectionState.value == ConnectionState.OFFLINE
+
+    override fun isConnecting(): Boolean = _connectionState.value == ConnectionState.CONNECTING
+
+    override fun isInitialized(): Boolean = _initialized.value
```

Comment:
Hey @leandroBorgesFerreira,

What do you think about converting these functions into properties as per [this topic](https://kotlinlang.org/docs/coding-conventions.html#functions-vs-properties)?